packages/template-retail-react-app/app/components/_app-config/index.jsx

```diff
@@ -66,13 +74,23 @@ AppConfig.restore = (locals = {}) => {
     apiConfig.parameters.siteId = site.id
 
     locals.api = new CommerceAPI({...apiConfig, locale: locale.id, currency})
+    locals.fillUrlTemplate = createUrlTemplate(
+        appConfig,
+        site.alias || site.id,
+        locale.id || locale
+    )
+    locals.site = site
+    locals.locale = locale.id
```

Comment:
Yes. It's safe to assume that the locale will be defined since `resolveLocaleFromUrl()` always returns a value: 
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/1b27d3e7bbe62fb9e57a3d0777b633a481e29cc2/packages/template-retail-react-app/app/utils/utils.js#L282-L286